### PR TITLE
feat(autofix): Keyword search only downloads repo once

### PR DESCRIPTION
**Pull request by jennmueng**
The keyword search tool will only download the repo into a temp directory once. You will now need to use basetools with the context manager:
```
with BaseTools(self.context) as tools:
```
the tmp dir and downloaded repo will be deleted at the closing of the context manager.

### CODE REVIEW

File: src/seer/automation/autofix/components/coding/component.py [hunk 1]
```diff
@@ -89,83 +89,86 @@ def _handle_missing_file_changes(
...
-                else None
-            ),
-            context=self.context,
-        )
+        with BaseTools(self.context) as tools:
+            agent = ClaudeAgent(
+                tools=tools.get_tools(),
+                config=AgentConfig(
+                    system_prompt=CodingPrompts.format_system_msg(), interactive=True
+                ),
```
Comment (roaga) [inline src/seer/automation/autofix/components/coding/component.py:97]: This whole file is a diff, but is it just this `as tools` line at the top that's new?

Comment (roaga) [reply src/seer/automation/autofix/components/coding/component.py:97]: Is there really no other way to implement this besides the `with` syntax? It's kind of annoying.

Could we, for example, store the tools in memory in the context? (I don't have strong ideas here)

Comment (jennmueng) [reply src/seer/automation/autofix/components/coding/component.py:97]: the context manager is the cleanest and surest way the files get cleaned up at the end. Any other method will require you to call tools.cleanup() which is arguably more annoying right

Comment (jennmueng) [reply src/seer/automation/autofix/components/coding/component.py:97]: (and also lead to potential filesystem leaks where we don't delete the repo and it becomes a problem)

Comment (roaga) [reply src/seer/automation/autofix/components/coding/component.py:97]: If it's in memory, wouldn't it just disappear when the context stops being used?

Comment (jennmueng) [reply src/seer/automation/autofix/components/coding/component.py:97]: How can you store files in memory

Comment (jennmueng) [reply src/seer/automation/autofix/components/coding/component.py:97]: The keyword search tool download the repo to perform a search right

Comment (roaga) [reply src/seer/automation/autofix/components/coding/component.py:97]: Oh I was misunderstanding how the tool works

**Review by roaga — APPROVED** [review]

File: src/seer/automation/autofix/components/coding/component.py (hunk 1 above)
Comment (roaga) [reply src/seer/automation/autofix/components/coding/component.py:97]: Codecov tho

**Pull request merged**